wordpress.yaml

```diff
@@ -1,7 +1,152 @@
 ---
 - name: Deploy Wordpress
   hosts: all
+  become: yes
+  vars_files:
+    - /vagrant/vars.yaml
+
   tasks:
+    - name: Update APT
+      apt:
+        update_cache: yes
+        cache_valid_time: 86400
+        upgrade: 'yes'        
+
+    - name: Install packages
+      apt:
+        name: 
+        - python3-mysqldb
+        - nginx
+        - mysql-server
+        - php-fpm
+        - php-mysql
+        - php-curl
+        - php-gd
+        - php-intl
+        - php-mbstring
+        - php-soap
+        - php-xml
+        - php-xmlrpc
+        - php-zip
+          
+    - name: Configure firewall
+      ufw:
+        rule: allow
+        name: "Nginx Full"
+        state: enabled
+    
+    - name:
+      template:
+        src: "/vagrant/nginx.conf.j2"
+        dest: "/etc/nginx/sites-available/{{ wp_conf_dir }}"
+
+    - name: Enable Nginx configuration
+      file:
+        src: "/etc/nginx/sites-available/{{ wp_conf_dir }}"  
+        dest: "/etc/nginx/sites-enabled/{{ wp_conf_dir }}"
+        state: link
+
+    - name: Unlink default Nginx
+      file:
+        path: "/etc/nginx/sites-enabled/default"
+        state: absent
+      notify: Restart Nginx  
+  
+    - name: Set root password in MYSQL
+      mysql_user:
+        name: root
+        password: "{{ mysql_password }}"
+        login_user: root
+        login_password: "{{ mysql_password }}"
+        login_unix_socket: "/var/run/mysqld/mysqld.sock"
+
+    - name: Remove all MYSQL anonymous users
+      mysql_user:
+        name: ''
+        host_all: yes
+        state: absent
+        login_user: root
+        login_password: "{{ mysql_password }}"
+  
+    - name: Remove MySQL test database
+      mysql_db:
+        name: test
+        state: absent
+        login_user: root
+        login_password: "{{ mysql_password }}"
+
+    - name: Create new database
+      mysql_db:
+        name: "wordpress"
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+
+    - name: Create WordPress user
+      mysql_user:
+        name: "wordpressuser"
+        password: "{{ wp_database_password }}"
+        priv: "*.*:ALL,GRANT"
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+    
+    - name: Check WordPress directory existance
```

Comment:
what is the reason for this check here?

Reply:
maybe if somehow provisioning gets corrupted or while doing it again, it won't create the directory again. however, I'm not sure if I can also check it with its state or not